x-pack/test/api_integration/apis/slos/get_slo.ts

```diff
@@ -0,0 +1,540 @@
+        enabled: true,
+        createdAt: getResponse.body.createdAt,
+        updatedAt: getResponse.body.updatedAt,
+        version: 2,
+        instanceId: '*',
+        summary: {
+          sliValue: 0.857143,
+          errorBudget: {
+            initial: 0.01,
+            consumed: 14.2857,
+            remaining: -13.2857,
+            isEstimated: false,
+          },
+          status: 'VIOLATED',
+        },
+      });
+
+      const response = await supertestAPI
+        .get(`/api/observability/slos`)
+        .set('kbn-xsrf', 'true')
+        .send()
+        .expect(200);
+    });
+
+    it('gets slo by id and calculates SLI - timeslices calendarAligned', async () => {
+      const id = await createSLO({
+        groupBy: '*',
+        timeWindow: {
+          duration: '1w',
+          type: 'calendarAligned',
+        },
+        budgetingMethod: 'timeslices',
+        objective: {
+          target: 0.99,
+          timesliceTarget: 0.95,
+          timesliceWindow: '10m',
+        },
+      });
+
+      await new Promise((resolve) => setTimeout(resolve, 120000));
+
+      const getResponse = await supertestAPI
+        .get(`/api/observability/slos/${id}`)
+        .set('kbn-xsrf', 'true')
+        .send()
+        .expect(200);
+
+      // expect summary transform to be created
+      expect(getResponse.body).eql({
+        name: 'Test SLO for api integration',
+        description: 'Fixture for api integration tests',
+        indicator: {
+          type: 'sli.kql.custom',
+          params: {
+            index: 'kbn-data-forge*',
+            filter: `system.network.name: eth1`,
+            good: 'container.cpu.user.pct < 1',
+            total: 'container.cpu.user.pct: *',
+            timestampField: '@timestamp',
+          },
+        },
+        budgetingMethod: 'timeslices',
+        timeWindow: { duration: '1w', type: 'calendarAligned' },
+        objective: {
+          target: 0.99,
+          timesliceTarget: 0.95,
+          timesliceWindow: '10m',
+        },
+        tags: ['test'],
+        groupBy: '*',
+        id,
+        settings: { syncDelay: '1m', frequency: '1m' },
+        revision: 1,
+        enabled: true,
+        createdAt: getResponse.body.createdAt,
+        updatedAt: getResponse.body.updatedAt,
+        version: 2,
+        instanceId: '*',
+        summary: {
+          sliValue: 0,
+          errorBudget: {
+            initial: 0.01,
+            consumed: 0.198413,
+            remaining: 0.801587,
+            isEstimated: false,
+          },
+          status: 'DEGRADING',
+        },
+      });
+
+      const response = await supertestAPI
+        .get(`/api/observability/slos`)
+        .set('kbn-xsrf', 'true')
+        .send()
+        .expect(200);
+    });
+
+    it('gets slos by query', async () => {
+      const id = await createSLO();
+      const secondId = await createSLO({ name: 'test name int' });
+
+      await new Promise((resolve) => setTimeout(resolve, 60000));
```

Comment:
@dominiqueclarke How about you create a `waitForSloCreated` method similar to what I have [here](https://github.com/elastic/kibana/pull/172786/files#diff-e214bd3b81ec89da711c2b4a2b852717e0738def0217e29683f9790d3895c4ebR77), that retries until SLO is created? If I understand correctly, with what you currently have, it will have to wait for 6 minutes, right? With a retry it might finish earlier, no? If you plan to use what I suggest, you will have to slightly modify and wait for both SLOs to be created.

Reply:
I changed this to retry.retryForTime which works a lot better.